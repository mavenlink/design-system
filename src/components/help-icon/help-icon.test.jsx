import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import HelpIcon from './help-icon.jsx';

describe('HelpIcon', () => {
  const requiredProps = {
    id: 'help-tooltip',
    label: 'Icon Label',
    text: 'Help Text',
  };

  it('has defaults', () => {
    render(<HelpIcon {...requiredProps} />);
    expect(document.body).toMatchSnapshot();

    user.hover(screen.getByTitle('Icon Label'));
    expect(document.body).toMatchSnapshot();
  });

  describe('hovering behavior', () => {
    it('does not render the tooltip unless the child is hovered', () => {
      render(<HelpIcon {...requiredProps} />);
      expect(screen.queryByText('Help Text')).not.toBeInTheDocument();
    });

    it('renders the tooltip when the icon is hovered over', () => {
      render(<HelpIcon {...requiredProps} />);
      user.hover(screen.getByTitle('Icon Label'));
      expect(screen.getByText('Help Text')).toBeInTheDocument();
    });

    it('removes the tooltip when the icon is no longer hovered', () => {
      render(<HelpIcon {...requiredProps} />);

      user.hover(screen.getByTitle('Icon Label'));
      expect(screen.getByText('Help Text')).toBeInTheDocument();

      user.unhover(screen.getByTitle('Icon Label'));
      expect(screen.queryByText('Help Text')).not.toBeInTheDocument();
    });
  });

  describe('direction api', () => {
    it('allows top', () => {
      render(<HelpIcon {...requiredProps} direction="top" />);
      expect(document.body).toMatchSnapshot();
    });

    it('allows bottom', () => {
      render(<HelpIcon {...requiredProps} direction="bottom" />);
      expect(document.body).toMatchSnapshot();
    });

    it('allows left', () => {
      render(<HelpIcon {...requiredProps} direction="left" />);
      expect(document.body).toMatchSnapshot();
    });

    it('allows right', () => {
      render(<HelpIcon {...requiredProps} direction="right" />);
      expect(document.body).toMatchSnapshot();
    });
  });
});
