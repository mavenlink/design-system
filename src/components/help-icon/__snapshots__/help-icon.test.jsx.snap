// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelpIcon direction api allows bottom 1`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <svg
        height="18"
        role="img"
        width="18"
      >
        <title>
          Icon Label
        </title>
        <use
          xlink:href="#help.svg"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`HelpIcon direction api allows left 1`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <svg
        height="18"
        role="img"
        width="18"
      >
        <title>
          Icon Label
        </title>
        <use
          xlink:href="#help.svg"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`HelpIcon direction api allows right 1`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <svg
        height="18"
        role="img"
        width="18"
      >
        <title>
          Icon Label
        </title>
        <use
          xlink:href="#help.svg"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`HelpIcon direction api allows top 1`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <svg
        height="18"
        role="img"
        width="18"
      >
        <title>
          Icon Label
        </title>
        <use
          xlink:href="#help.svg"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`HelpIcon has defaults 1`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <svg
        height="18"
        role="img"
        width="18"
      >
        <title>
          Icon Label
        </title>
        <use
          xlink:href="#help.svg"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`HelpIcon has defaults 2`] = `
<body>
  <div>
    <div
      class="wrapper"
    >
      <svg
        height="18"
        role="img"
        width="18"
      >
        <title>
          Icon Label
        </title>
        <use
          xlink:href="#help.svg"
        />
      </svg>
    </div>
  </div>
  <span
    class="tooltip top"
    id="help-tooltip"
    style="left: 0px; top: -128px;"
  >
    Help Text
  </span>
</body>
`;
